core/src/main/java/com/linecorp/armeria/client/endpoint/dns/DnsEndpointGroup.java

```diff
@@ -0,0 +1,359 @@
+     * @param defaultPort the port to use when the DNS answer does not contain one. Generally the port is only
+     *     available in the answer when {@code dnsType} is {@link DnsRecordType#SRV}. {@code 0} indicates an
+     *     unspecified port, meaning the port will either come from the DNS answer or use a protocol-specific
+     *     well-defined port number.
+     * @param eventLoop the {@link EventLoop} to schedule DNS queries on.
+     */
+    public static DnsEndpointGroup of(
+            String hostname, DnsRecordType dnsType, int defaultPort, EventLoop eventLoop) {
+        return DnsEndpointGroup.of(hostname, dnsType, defaultPort, eventLoop, Duration.ofSeconds(1));
+    }
+
+    /**
+     * Creates a {@link DnsEndpointGroup}.
+     *
+     * @param hostname the hostname to query DNS queries for.
+     * @param dnsType the DNS query type.
+     * @param defaultPort the port to use when the DNS answer does not contain one. Generally the port is only
+     *     available in the answer when {@code dnsType} is {@link DnsRecordType#SRV}. {@code 0} indicates an
+     *     unspecified port, meaning the port will either come from the DNS answer or use a protocol-specific
+     *     well-defined port number.
+     * @param eventLoop the {@link EventLoop} to schedule DNS queries on.
+     * @param queryInterval the {@link Duration} to query DNS at.
+     */
+    public static DnsEndpointGroup of(
+            String hostname, DnsRecordType dnsType, int defaultPort, EventLoop eventLoop,
+            Duration queryInterval) {
+        return new DnsEndpointGroup(hostname, dnsType, defaultPort, createResolverForEventLoop(eventLoop),
+                                    eventLoop, queryInterval);
+    }
+
+    private static final Logger logger = LoggerFactory.getLogger(DnsEndpointGroup.class);
+
+    private static final int INADDRSZ4 = 4;
+    private static final int INADDRSZ6 = 16;
+
+    private final String hostname;
+    private final int defaultPort;
+    private final DnsRecordType recordType;
+    private final DnsNameResolver resolver;
+    private final EventLoop eventLoop;
+    private final Duration queryInterval;
+
+    private ScheduledFuture<?> scheduledFuture;
+
+    @VisibleForTesting
+    DnsEndpointGroup(String hostname, DnsRecordType recordType, int defaultPort, DnsNameResolver resolver,
+                     EventLoop eventLoop, Duration queryInterval) {
+        this.hostname = hostname;
+        this.defaultPort = defaultPort;
+        this.recordType = recordType;
+        this.resolver = resolver;
+        this.eventLoop = eventLoop;
+        this.queryInterval = queryInterval;
+    }
+
+    /**
+     * Starts polling for service updates.
+     */
+    public void start() {
+        checkState(scheduledFuture == null, "already started");
+        scheduledFuture = eventLoop.scheduleAtFixedRate(this::query, 0, queryInterval.getSeconds(),
+                                                        TimeUnit.SECONDS);
```

Comment:
The query interval could be adjusted dynamically based on the TTL value in the answer?